### PR TITLE
Music Player - updating onload management of track availability

### PR DESCRIPTION
**Pull request by iisa**
**Description**
YouTube wrapper behavior fine tuning.
This PR ensures the following:
1) On page load, changing channels does not update track nor autoplay
2) On page load, changing channel to YouTube, clicking on the selected track plays the video

#1 & 2) Can be tested on:
`https://www-isa.archive.org/details/cd_backstreet-boys_backstreet-boys/disc1/03.+Backstreet+Boys+-+Get+Down+(You're+The+One+For+Me).flac`
&&
`https://www-isa.archive.org/details/cd_backstreet-boys_backstreet-boys`
Repro Steps:
- load page
- change channel to YouTube
- change channel back to Internet Archive
  - basically, any channel change on load should NOT pollute URL. should not autoplay YouTube.
- NOW, change back to YouTube
- click on highlighted track,
- video will play

### CODE REVIEW

File: packages/ia-components/sandbox/audio-players/youtube-player.jsx [hunk 1]
```diff
@@ -53,8 +53,12 @@ class YoutubePlayer extends Component {
    * Update component only if there is a change in video id
    * If above update video id and selected track
    */
-  shouldComponentUpdate(nextProps) {
-    const { id } = this.state;
+  shouldComponentUpdate(nextProps, nextState) {
```
Comment (shaneriley) [inline packages/ia-components/sandbox/audio-players/youtube-player.jsx]: The comment above this method indicates that it processes to the end only if the video ID has changed, however it looks like it's only if the player prop has changed. Is that still correct?

File: packages/ia-components/sandbox/audio-players/youtube-player.jsx [hunk 2]
```diff
@@ -71,7 +75,10 @@ class YoutubePlayer extends Component {
    */
   componentDidUpdate() {
     const { id, player } = this.state;
-    player.loadVideoById(id, 'default');
+    const { id: propsID } = this.props;
+    const idToUse = id || propsID;
```
Comment (shaneriley) [inline packages/ia-components/sandbox/audio-players/youtube-player.jsx]: **Suggestion**: for consistency we can also pull the state ID as `stateID` to completely disambiguate the locals.

File: packages/ia-components/sandbox/audio-players/youtube-player.jsx [hunk 3]
```diff
@@ -92,23 +99,25 @@ class YoutubePlayer extends Component {
...
-          rel: 0,
-          enablejsapi: 1
-        },
-        events: {
-          onReady: this.onPlayerReady,
-          onStateChange: this.onPlayerStateChange,
-          onError: this.onPlayerError
-        }
-      })
+    const { id: propsID } = this.props;
```
Comment (shaneriley) [inline packages/ia-components/sandbox/audio-players/youtube-player.jsx]: Same suggestion here RE: stateID

File: packages/ia-components/sandbox/audio-players/youtube-player.jsx [hunk 4]
```diff
@@ -92,23 +99,25 @@ class YoutubePlayer extends Component {
...
+    const idToUse = id || propsID;
```
Comment (shaneriley) [inline packages/ia-components/sandbox/audio-players/youtube-player.jsx]: **Suggestion**: rename to videoId to take advantage of property definition shorthand.

File: packages/ia-components/sandbox/audio-players/youtube-player.jsx (hunk 1 above)
Comment (iisa) [reply packages/ia-components/sandbox/audio-players/youtube-player.jsx]: it still stands. thanks for checking
